scripts/task_process/task_proc_wrapper.sh

```diff
@@ -4,6 +4,29 @@ function cache_status {
     python task_process/cache_status.py
 }
 
+function check_exit {
+    ARG_DAG_STATUS=$1
+    ARG_ENTERED_CUR_STATUS=$2
+    # Checks if the TP can exit without losing any possible future updates to the status of the task
+    # First checks that the DAG is in a final state
+    # If that passes, checks if the dag has been in a final state for long enough (currently 24h)
+    # It will echo 1 if TP can exit, otherwise 0
+
+    if [[ "$ARG_DAG_STATUS" != "1" && "$ARG_DAG_STATUS" != "2" && "$ARG_DAG_STATUS" != "init" && "$ARG_DAG_STATUS" ]]; then
```

Comment:
please put a comment saying what those status number mean. As a help to who will maintain this code in the future. May also want to spend one line to explain "init" vs. "1", "2" etc.